### PR TITLE
Check VPN connection as part of system components condition

### LINKED ISSUE #1167: Gardener reconcile/readiness shoot status doesn't reflect the working VPN
**What happened**:

If the shoot cluster VPN is not reachable and the `vpn-seed` container inside the `kube-apiserver` pod produces error logs:

```
[Wed Jul  3 14:59:19 UTC 2019]: trying to identify the endpoint (load balancer name of vpn-shoot service) myself...
[Wed Jul  3 14:59:19 UTC 2019]: found endpoints: [ 10.0.0.99 ]
[Wed Jul  3 14:59:19 UTC 2019]: checking whether port 4314 is open on 10.0.0.99 ...
[Wed Jul  3 14:59:22 UTC 2019]: error: port 4314 on 10.0.0.99 is not open, can not use it
[Wed Jul  3 14:59:22 UTC 2019]: No tunnel endpoint found
```

This error is not reflected in the overall shoot cluster status:

```
$ kubectl -n garden-cluster get shoot
NAME         CLOUDPROFILE   VERSION   SEED           DOMAIN                                 OPERATION   PROGRESS   APISERVER   CONTROL   NODES   SYSTEM   AGE
wir5xvwkm5   region1        1.15.0    seed-region1   wir5xvwkm5.rtr-prblm.shoot.my.domain   Succeeded   100        True        True      True    True     3h47m
```

And when user want to use `kubectl logs` or `kubectl exec`, this operation will timeout and user won't be able to understand why.

**What you expected to happen**:

I expect to see PROGRESS: 99%, SYSTEM: false, OPERATION: Creating. In addition the gardener dashboard should display the VPN connection issue with logs inside the status modal window.

**Environment**:

- Gardener version: 0.26.1
- Kubernetes version (use `kubectl version`): 
- Cloud provider or hardware configuration: openstack
- Others: dashboard 1.33.0

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
As part of the health checks for the `SystemComponentsReady` condition we are now also checking whether the VPN connection has been established.

**Which issue(s) this PR fixes**:
Fixes #1167

**Special notes for your reviewer**:
Example how a broken VPN connection is displayed:

```yaml
conditions:
...
- lastTransitionTime: "2020-02-07T13:22:27Z"
  lastUpdateTime: "2020-02-07T13:23:04Z"
  message: 'VPN connection has not been established (could not forward to vpn-shoot
    pod: port forward connection could not be established within five seconds)'
  reason: VPNConnectionBroken
  status: Progressing
  type: SystemComponentsHealthy
```

**Release note**:

```improvement user
The `SystemComponentsHealthy` condition now also reflects the status of the connectivity between the control plane and the worker nodes.
```

### CODE REVIEW

**Review by timuthy — DISMISSED** [review]
/lgtm

File: pkg/operation/botanist/health_check.go [hunk 1]
```diff
@@ -615,6 +617,7 @@ func (b *Botanist) checkSystemComponents(
 	shootDaemonSetLister kutil.DaemonSetLister,
 	extensionConditions []extensionCondition,
 ) (*gardencorev1beta1.Condition, error) {
+	ctx := context.TODO()
```
Comment (timebertt) [inline pkg/operation/botanist/health_check.go]: Can we add the context as the first parameter instead of creating one in here?

**Review by timebertt — CHANGES_REQUESTED** [review]
Generally /lgtm, nice PR
Just one small suggestion..

File: pkg/operation/botanist/health_check.go (hunk 1 above)
Comment (rfranzke) [reply pkg/operation/botanist/health_check.go]: Done, although it doesn't help too much. The whole file is not used `context` yet, and there must be dedicated time spent to implement it cleanly (#1648)

Comment (timebertt) [reply pkg/operation/botanist/health_check.go]: Ok, got it. Thanks, anyways.

**Review by timebertt — APPROVED** [review]
/approve

**Review by timuthy — APPROVED** [review]

**Pull request merged**